WcaOnRails/app/models/competition.rb

```diff
@@ -622,6 +622,62 @@ def continent
     country ? Continent.find_by_id(country.continentId) : nil
   end
 
+  def psych_sheet_event(event)
+    preferred_format = event.preferred_formats.first
+    sort_clause = (preferred_format.sort_by == :single) ? "-single_rank desc, -average_rank desc, users.name" : "-average_rank desc, -single_rank desc, users.name"
```

Comment:
Two thoughts:
- logically would be to define this variable below `joinsql` and `selectsql`
- you can avoid using ternary by doing
```ruby
sort_clause = "-#{preferred_format.sort_by} desc, -#{preferred_format.sort_by_second}_rank desc, users.name"
```